tests/src/Rules/data/bug-437.php

```diff
@@ -0,0 +1,20 @@
+<?php
+
+namespace Bug437;
+
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\Core\Entity\Query\QueryInterface;
+
+final class Foo {
+    private QueryInterface $vocabularyQuery;
+    public function __construct(EntityTypeManagerInterface $entityTypeManager)
+    {
+        $this->vocabularyQuery = $entityTypeManager
+            ->getStorage('taxonomy_term')
+            ->getQuery()
+            ->accessCheck(TRUE);
+    }
+    public function a(): int {
+        return $this->vocabularyQuery->count()->execute();
+    }
```

Comment:
Where this seems to break down:

```
            $returnType = new EntityQueryCountType(
                $varType->getClassName(),
                $varType->getSubtractedType(),
                $varType->getClassReflection()
            );
            if ($varType instanceof EntityQueryType && $varType->hasAccessCheck()) {
                return $returnType->withAccessCheck();
            }
```

In `\mglaman\PHPStanDrupal\Type\EntityQuery\EntityQueryDynamicReturnTypeExtension::getTypeFromMethodCall`. Previously it was set to be an access checked object type. But I believe the property typehint overrides what our dynamic extension return type has done.